### PR TITLE
Accept also Mongo Collection as param for MongodbSender

### PR DESCRIPTION
**Pull request by arggh**
This changes the handling of collection argument in MongodbSender constructor, which now accepts either:

a) a collection name, which will then be instantiated as previously
b) an already existing Collection, which will be used instead

### CODE REVIEW

File: src/Senders/MongodbSender.js [hunk 1]
```diff
@@ -6,16 +6,23 @@ export default class MongodbSender extends SenderBase {
    *
    * @param {Mongo} mongo
    *   The Meteor Mongo service.
-   * @param {String} collectionName
-   *   The name of the collection in which to log.
+   * @param {(String|Collection)} Collection or collection name
```
Comment (FGM) [inline src/Senders/MongodbSender.js]: This is the name of the variable, so just "collection"

File: src/Senders/MongodbSender.js [hunk 2]
```diff
@@ -6,16 +6,23 @@ export default class MongodbSender extends SenderBase {
...
    */
-  constructor(mongo, collectionName = 'logger') {
+  constructor(mongo, collection = 'logger') {
     super();
     this.mongo = mongo;
-    const collection = new mongo.Collection(collectionName);
-    this.store = collection
-      ? collection
-      : new this.mongo.Collection(collectionName);
+    if (typeof collection === 'object') {
```
Comment (FGM) [inline src/Senders/MongodbSender.js]: Is it reasonable to accept any object ? Or should we be stricter ?

File: src/Senders/MongodbSender.js [hunk 3]
```diff
@@ -6,16 +6,23 @@ export default class MongodbSender extends SenderBase {
...
+   *   The collection or the name of the collection in which to log.
    */
-  constructor(mongo, collectionName = 'logger') {
+  constructor(mongo, collection = 'logger') {
     super();
     this.mongo = mongo;
-    const collection = new mongo.Collection(collectionName);
-    this.store = collection
-      ? collection
-      : new this.mongo.Collection(collectionName);
```
Comment (FGM) [inline src/Senders/MongodbSender.js]: Actually, I think your changes pinpointed a bug here: the goal of this code is already to reuse a collection if one has been set, but that's not what the code does.

Comment (FGM) [inline src/Senders/MongodbSender.js]: The idea was to create a new collection if needed, or reuse the existing singleton collection, but it seems not to be doing what it expects to do.

**Comment by FGM** [comment]
Added tests, coding style fixes, ignored yarn.lock. Merge to master, thanks.

**Comment by arggh** [comment]
Sorry for taking my time with this, I was super busy with work.

**Pull request closed without merging**